esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp

```diff
@@ -949,6 +959,96 @@ bool CWsESDLConfigEx::onConfigureESDLBindingMethod(IEspContext &context, IEspCon
+
+        StringBuffer username;
+        context.getUserID(username);
+        DBGLOG("CWsESDLConfigEx::onConfigureESDBindingLogTransform User=%s",username.str());
+
+        const char* bindingId = req.getEsdlBindingId();
+        const char* logTransformName = req.getLogTransformName();
+        StringBuffer espProcName;
+        StringBuffer espBindingName;
+        StringBuffer esdlServiceName;
+        StringBuffer esdlDefIdSTR;
+        StringBuffer config(req.getConfig());
+        if (config.isEmpty())
+            throw MakeStringException(-1, "Config not defined.");
+
+        StringBuffer msg;
+        Owned<IPropertyTree> bindingtree = m_esdlStore->getBindingTree(bindingId, msg);
+        if(!bindingtree)
+            throw MakeStringException(-1, "Can't find esdl binding for id %s", bindingId);
+
+        bindingtree->getProp("@espprocess", espProcName);
+        bindingtree->getProp("@espbinding", espBindingName);
+        bindingtree->getProp("Definition[1]/@esdlservice", esdlServiceName);
+        bindingtree->getProp("Definition[1]/@id", esdlDefIdSTR);
+
+        Owned<IPropertyTree> logTransformTree = fetchConfigInfo(config, bindingId, ESDLLogTransform);
+        if (!logTransformTree || logTransformTree->getCount("LogTransform") <= 0)
+            throw MakeStringException(-1, "Could not find any LogTransform configuration entries.");
+
+        const char *esdlDefId = esdlDefIdSTR.str();
+        if (isEmptyString(esdlDefId))
+            throw MakeStringException(-1, "Can't find esdl definition for binding %s", bindingId);
+
+        StringBuffer esdlDefinitionName;
+        while (esdlDefId && *esdlDefId != '.')
+            esdlDefinitionName.append(*esdlDefId++);
+
+        if (isEmptyString(esdlDefId))
+            throw MakeStringException(-1, "Invalid ESDL Definition ID format detected: '%s'. Expected format: <esdldefname>.<ver>", esdlDefIdSTR.str());
+
+        int esdlver = 0;
+        esdlDefId++;
+        if (esdlDefId)
+            esdlver = atoi(esdlDefId);
+
+        if (esdlver <= 0)
+            throw MakeStringException(-1, "Invalid ESDL Definition version detected: %d", esdlver);
+
+        bool override = req.getOverwrite();
+        if (m_esdlStore->definitionExists(esdlDefIdSTR.str()))
+        {
+            StringBuffer logTransformXPath;
+            if (!isEmptyString(logTransformName))
+                logTransformXPath.appendf("LogTransform[@name='%s']", logTransformName);
+            else
+                logTransformXPath.append("LogTransform");
+            Owned<IPropertyTreeIterator> iter = logTransformTree->getElements(logTransformXPath.str());
+            ForEach(*iter)
+            {
+                IPropertyTree &item = iter->query();
+                StringBuffer msg;
+                success = m_esdlStore->configureLogTransform(bindingId, logTransformName, LINK(&item), override, msg);
+                resp.updateStatus().setDescription(msg.str());
```

Comment:
Would it be better to accumulate the results of each iteration instead of allowing the last iteration to override the results of all prior?

Reply:
yes